cpp2sky/segment_context.h

```diff
@@ -208,6 +204,19 @@ class SegmentContext {
  public:
   virtual ~SegmentContext() = default;
 
+  /**
+   * Change sampling flag. When this value is true, belonging spans
+   * will be sent to OAP. This value is inconfigurable when self segment context
+   * is not root because it will be specified by propagated flag.
+   */
+  virtual void setDefaultSamplingStatus(bool do_sample) = 0;
```

Comment:
Just FYI, all other language agents are not process this sampling flag. Once the `sw8` header found, the tracing will continue. Just consider the `analysis` flag when you report data. When we activate observability of service mesh, the tracing could be `no-analysis` mode(optional).

Reply:
Does it mean that `sampling` flag on propagation header has no mean actually, right? `skywalking-python` seems that this value is always true. https://github.com/apache/skywalking-python/blob/master/skywalking/trace/carrier.py#L67
So we should change send span or not can be determined at `skipAnalysis`, right?

Reply:
> Does it mean that sampling flag on propagation header has no mean actually, right?

Currently, yes. Only header existence matters.

> So we should change send span or not can be determined at skipAnalysis, right?

No, we don't. The span should be sent always. This flag is telling the backend, we do want to trace only, but don't need metrics/topology analyzed from them.